packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1610,13 +1615,30 @@ export class Bucket extends BucketBase {
     }
     Bucket.validateBucketName(bucketName);
 
+    const _regionInfo = regionInfo.RegionInfo.get(region);
+    const oldUrlFormatRegions = new Set(regionsBefore(RULE_S3_WEBSITE_REGIONAL_SUBDOMAIN));
```

Comment:
This is knowledge I'd really like to keep inside the `region-info` module. It shouldn't be replicated here. Make a helper function there if you need to.

But I'm pretty sure this or something similar to it is already in that module. Why are we duplicating it here?

Reply:
`regionsBefore` has not been used yet, so I created a new helper method in `RegionInfo` 🙌